File: frontend/src/types/payment.ts
/**
 * Payment System Type Definitions
 */

// ==================== Status & Type Literals ====================

export type PaymentStatus =
  | 'pending'
  | 'paid'
  | 'completed'
  | 'failed'
  | 'cancelled'
  | 'expired'
  | 'refunded'
  | 'partially_refunded'
  | 'refund_pending'

export type PaymentType =
  | 'alipay'
  | 'wechat'
  | 'stripe'
  | 'usdt'
  | 'balance'

export type OrderType =
  | 'recharge'
  | 'balance'
  | 'subscription'

// ==================== Core Entities ====================

export interface PaymentOrder {
  id: number
  user_id: number
  user_email: string
  user_name: string
  amount: number
  pay_amount: number
  fee_rate: number
  recharge_code: string
  status: PaymentStatus
  payment_type: PaymentType
  payment_trade_no: string
  pay_url: string
  qr_code: string
  qr_code_img: string
  refund_amount: number
  refund_reason: string
  refund_at: string | null
  force_refund: boolean
  refund_requested_at: string | null
  refund_request_reason: string
  expires_at: string | null
  paid_at: string | null
  completed_at: string | null
  failed_at: string | null
  failed_reason: string
  client_ip: string
  src_host: string
  src_url: string
  order_type: OrderType
  plan_id: number | null
  subscription_group_id: number | null
  subscription_days: number | null
  provider_instance_id: number | null
  created_at: string
  updated_at: string
}

export interface MethodLimit {
  paymentType: PaymentType
  available: boolean
  dailyLimit: number
  dailyUsed: number
  remaining: number
  singleMin: number
  singleMax: number
  feeRate: number
}

export interface PaymentConfig {
  settings: Record<string, string>
  limits: MethodLimit[]
}

export interface PaymentChannel {
  id: number
  group_id: number | null
  name: string
  platform: string
  rate_multiplier: number
  description: string
  models: string[]
  features: string[]
  sort_order: number
  enabled: boolean
  created_at: string
  updated_at: string
}

export interface SubscriptionPlan {
  id: number
  group_id: number | null
  name: string
  description: string
  price: number
  original_price: number
  validity_days: number
  validity_unit: string
  features: string[]
  product_name: string
  for_sale: boolean
  sort_order: number
  created_at: string
  updated_at: string
}

export interface PaymentAuditLog {
  id: number
  order_id: number
  action: string
  detail: string
  operator: string
  created_at: string
}

export interface ProviderInstance {
  id: number
  provider_key: string
  name: string
  config: Record<string, unknown>
  supported_types: PaymentType[]
  enabled: boolean
  sort_order: number
  limits: Record<string, unknown>
  refund_enabled: boolean
  created_at: string
  updated_at: string
}

// ==================== Dashboard ====================

export interface PaymentDashboardStats {
  today_amount: number
  today_order_count: number
  total_amount: number
  total_order_count: number
  daily_series: { date: string; amount: number; count: number }[]
  payment_methods: { type: PaymentType; amount: number; count: number }[]
  leaderboard: { user_id: number; user_email: string; total_amount: number; order_count: number }[]
}

// ==================== Request / Response ====================

export interface CreateOrderRequest {
  amount: number
  payment_type: PaymentType
  order_type: OrderType
  plan_id?: number
  return_url?: string
  src_host?: string
  src_url?: string
}

export interface CreateOrderResponse {
  order: PaymentOrder
  pay_url: string
  qr_code: string
  client_secret: string
  access_token: string
}

File: backend/internal/service/payment.go
package service

import (
	"time"

	"github.com/shopspring/decimal"
)

// PaymentOrder is the service-layer domain model for a payment order.
type PaymentOrder struct {
	ID                  int64
	UserID              int64
	UserEmail           *string
	UserName            *string
	UserNotes           *string
	Amount              decimal.Decimal
	PayAmount           *decimal.Decimal
	FeeRate             *decimal.Decimal
	RechargeCode        string
	Status              string
	PaymentType         string
	PaymentTradeNo      *string
	PayURL              *string
	QrCode              *string
	QrCodeImg           *string
	RefundAmount        *decimal.Decimal
	RefundReason        *string
	RefundAt            *time.Time
	ForceRefund         bool
	RefundRequestedAt   *time.Time
	RefundRequestReason *string
	RefundRequestedBy   *int64
	ExpiresAt           time.Time
	PaidAt              *time.Time
	CompletedAt         *time.Time
	FailedAt            *time.Time
	FailedReason        *string
	ClientIP            *string
	SrcHost             *string
	SrcURL              *string
	OrderType           string
	PlanID              *int64
	SubscriptionGroupID *int64
	SubscriptionDays    *int
	ProviderInstanceID  *int64
	CreatedAt           time.Time
	UpdatedAt           time.Time
}

// PaymentAuditLog is the service-layer domain model for payment audit logs.
type PaymentAuditLog struct {
	ID        int64
	OrderID   int64
	Action    string
	Detail    *string
	Operator  *string
	CreatedAt time.Time
}

// PaymentChannel is the service-layer domain model for a payment channel display config.
type PaymentChannel struct {
	ID             int64
	GroupID        *int64
	Name           string
	Platform       string
	RateMultiplier decimal.Decimal
	Description    *string
	Models         *string
	Features       *string
	SortOrder      int
	Enabled        bool
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

// PaymentProviderInstance is the service-layer domain model for a provider instance.
type PaymentProviderInstance struct {
	ID             int64
	ProviderKey    string
	Name           string
	Config         string // AES-256-GCM encrypted JSON
	SupportedTypes string // comma-separated PaymentType values
	Enabled        bool
	SortOrder      int
	Limits         *string // JSON: per-type daily/single limits
	RefundEnabled  bool
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

// SubscriptionPlan is the service-layer domain model for a subscription plan.
type SubscriptionPlan struct {
	ID            int64
	GroupID       *int64
	Name          string
	Description   *string
	Price         decimal.Decimal
	OriginalPrice *decimal.Decimal
	ValidityDays  int
	ValidityUnit  string
	Features      *string
	ProductName   *string
	ForSale       bool
	SortOrder     int
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

// ProviderInstanceLimits represents parsed per-type limits for a provider instance.
type ProviderInstanceLimits struct {
	DailyLimit *decimal.Decimal `json:"daily_limit,omitempty"`
	SingleMin  *decimal.Decimal `json:"single_min,omitempty"`
	SingleMax  *decimal.Decimal `json:"single_max,omitempty"`
}
